runtime/src/drivers.rs

```diff
@@ -78,34 +80,23 @@ impl Drivers {
     /// Callers must ensure that this function is called only once, and that
     /// any concurrent access to these register blocks does not conflict with
     /// these drivers.
-    pub unsafe fn new_from_registers() -> CaliptraResult<Self> {
-        let mut trng = Trng::new(
-            CsrngReg::new(),
-            EntropySrcReg::new(),
-            SocIfcTrngReg::new(),
-            &SocIfcReg::new(),
-        )?;
-
-        let mut sha384 = Sha384::new(Sha512Reg::new());
-        let mut ecc384 = Ecc384::new(EccReg::new());
-        let mut hmac384 = Hmac384::new(HmacReg::new());
-        let mut key_vault = KeyVault::new(KvReg::new());
+    pub fn new_from_registers() -> CaliptraResult<Self> {
```

Comment:
Aren't these a bit orthogonal since unsafe doesn't necessarily imply that the function can't be called more than once? We already have the comment on new_from_registers telling us about safety.

Reply:
They're technically orthogonal, it's just that calling an unsafe function requires you to write extra code to use it, so the idea is that people will be more careful with functions marked unsafe.

Reply:
Done